### PR TITLE
PA-1554, PA-2366

### PR DESCRIPTION
**Pull request by devinleighsmith**
Only display the 'active' or 'selected' marker during initial load.
updates to draft markers will now cause the map to jump to the most recently edited draft marker.

### CODE REVIEW

File: frontend/src/components/maps/BuildingPopupView.tsx [hunk 1]
```diff
@@ -92,6 +92,7 @@ export const BuildingPopupView: React.FC<IBuildingDetailProps> = (props: IBuildi
                       disabled: true,
                       loadDraft: false,
                       buildingId: buildingDetail?.id,
+                      parcelId: undefined,
```
Comment (devinleighsmith) [inline frontend/src/components/maps/BuildingPopupView.tsx:95]: always set the unused id to undefined.

File: frontend/src/components/maps/leaflet/InventoryLayer.tsx [hunk 2]
```diff
@@ -94,18 +94,11 @@ export const InventoryLayer: React.FC<InventoryLayerProps> = ({
       (filter: IGeoSearchParams) => {
         loadProperties(filter)
           .then(async (data: Feature[]) => {
-            const points = data
-              .filter(feature => {
```
Comment (devinleighsmith) [inline frontend/src/components/maps/leaflet/InventoryLayer.tsx:98]: no longer need this funky filter to remove the matching property.

File: frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx [hunk 3]
```diff
@@ -51,12 +48,6 @@ export const saveParcelDataLayerResponse = (
         },
       }),
     );
-    //save the entire feature to redux for use within the map.
```
Comment (devinleighsmith) [inline frontend/src/components/maps/leaflet/LayerPopup/hooks/useLayerQuery.tsx:54]: Not required anymore.

File: frontend/src/components/maps/leaflet/PointClusterer.tsx [hunk 4]
```diff
@@ -68,6 +73,22 @@ export const PointClusterer: React.FC<PointClustererProps> = ({
     zoom,
     options: { radius: 60, extent: 256, minZoom, maxZoom },
   });
+  const currentClusterIds = useMemo(() => {
```
Comment (devinleighsmith) [inline frontend/src/components/maps/leaflet/PointClusterer.tsx:76]: this gets the list of pims property ids, from the active cluster.

File: frontend/src/components/maps/providers/FIlterProvider.tsx [hunk 5]
```diff
@@ -11,8 +11,8 @@ const FilterContext = React.createContext<{
  * helps the inventory layer zoom to the results when submitting the filter form
  */
 export const FilterProvider: React.FC = ({ children }) => {
-  // Default changed state to true on page load
-  const [changed, setChanged] = React.useState(true);
+  // Default changed state to false on page load
+  const [changed, setChanged] = React.useState(false);
```
Comment (devinleighsmith) [inline frontend/src/components/maps/providers/FIlterProvider.tsx:15]: Needed this to keep the initially opened popup centered properly.

File: frontend/src/components/maps/leaflet/SelectedPropertyMarker/SelectedPropertyMarker.tsx [hunk 6]
```diff
@@ -1,18 +1,20 @@
...
 /**
  * Wrapper of the React Leaflet marker to auto open the popup for a selected property
  */
 const SelectedPropertyMarker: React.FC<MarkerProps & { map: Map }> = props => {
   const ref = React.useRef<any>(undefined);
 
-  React.useEffect(() => {
+  useDeepCompareEffect(() => {
     if (ref.current.leafletElement) {
+      props.map.setView(props.position, props.map.getZoom());
```
Comment (devinleighsmith) [inline frontend/src/components/maps/leaflet/SelectedPropertyMarker/SelectedPropertyMarker.tsx:14]: center the selected popup when opened.

**Review by Fosol — APPROVED** [review]

File: frontend/src/components/maps/leaflet/InventoryLayer.tsx [hunk 7]
```diff
@@ -94,18 +94,11 @@ export const InventoryLayer: React.FC<InventoryLayerProps> = ({
...
-                  feature.properties!.propertyTypeId === selected?.propertyTypeId &&
-                  feature.properties!.id === selected?.parcelDetail?.id
-                );
-              })
-              .map(f => {
-                return {
-                  ...f,
-                } as PointFeature;
-              });
+            const points = data.map(f => {
```
Comment (goemen) [inline frontend/src/components/maps/leaflet/InventoryLayer.tsx:97]: This map tranform is not necessary

**Review by goemen — APPROVED** [review]
Minor tings, but looks good

**Pull request merged**